# box filter thresholds(1209)
#x_min: -1 # m
#x_max: 1
#y_min: -3
#y_max: 0
#z_min: 0

# box filter thresholds(try to be looser)
x_min: -1.02 # m
x_max: 1.02
y_min: -3.1
y_max: 0
z_min: 0

# brightness filter thresholds
brightnessStrictness: 0.9 # (0,1), the bigger, the more strict
# LED Separation Strictness
# for red ring
RStrcitnessMax: 0.5 # (0,1), the smaller, the more strict
RStrictnessMin: 0.5 # (0,1), the bigger, the more strict
# for green ring
GStrcitnessMax: 0.5 # (0,1), the smaller, the more strict
GStrictnessMin: 0.5 # (0,1), the bigger, the more strict

# noise-removal filter thresholds(!!!!!!!!!!to be written to Config)
min_valid_vel: 0.05 # m/s
max_valid_y_vel: 0.05 # m/s
min_LED_dis: 0.3 # m
max_LED_dis: 0.7 # m

# finish criteria
finishPos: [-0.1,-0.2,1.5708]
finishThreshold: 0.14

# topics
rgb_topic: /camera/color/image_raw
depth_topic: /camera/aligned_depth_to_color/image_raw
raw_cloud_topic: /camera/depth/color/points
calied_cloud_topic: /calied_cloud
boat_cloud_topic: /boat_cloud
red_cloud_topic: /red_cloud
green_cloud_topic: /green_cloud
localization_topic: /localization
boat_vel_vis_topic: /boat_vel_vis
to_control_topic: /to_control

# subpaths
rgb_subpath: rgb
depth_subpath: depth
rosbag_subpath: rosbag

# T_camera_cali_MATLAB
# dorm_floor
#R_MATLAB: [0.9987,-0.0002,-0.0509, -0.0453,0.4514,-0.8912, 0.0232,0.8923,0.4508]
#p_MATLAB: [-0.0482, 0.2211, 1.8816]
#correction: [0, 0.5, 0.25, 0, 0, -0.3491] # trans xyz,rot zyx; rotate first and then translate
#distortion_corr: [1, -3, -0.57735] # [x0,y0,k] undistorted (x,y,z) will distort into (x, y+k(x-x0)*(y/y0), z)

# 用于pool1121_demo数据集
#R_MATLAB: [0.9997,0.0181,-0.0139,-0.0217,0.9422,-0.3344,0.0070,0.3346,0.9423]
#p_MATLAB: [-0.0705,-0.2156,0.6126]
#correction: [-0.1,0,-0.8,0,0,0.55] # trans xyz,rot zyx; rotate first and then translate
#distortion_corr: [1, -3, -0.57735] # [x0,y0,k] undistorted (x,y,z) will distort into (x, y+k(x-x0)*(y/y0), z)

# 用于1203数据集
#R_MATLAB: [0.9997,0.0181,-0.0139,-0.0217,0.9422,-0.3344,0.0070,0.3346,0.9423]
#p_MATLAB: [-0.0705,-0.2156,0.6126]
#correction: [-0.05,0,-0.9,0.05,0,0.5] # trans xyz,rot zyx; rotate first and then translate
#distortion_corr: [1, -3, -0.57735] # [x0,y0,k] undistorted (x,y,z) will distort into (x, y+k(x-x0)*(y/y0), z)

# pool1205 success
R_MATLAB: [0.9997,0.0181,-0.0139,-0.0217,0.9422,-0.3344,0.0070,0.3346,0.9423]
p_MATLAB: [-0.0705,-0.2156,0.6126]
correction: [-0.05,0.05,-0.96,0.04,0,0.5] # trans xyz,rot zyx; rotate first and then translate
distortion_corr: [1, -3, -0.57735] # [x0,y0,k] undistorted (x,y,z) will distort into (x, y+k(x-x0)*(y/y0), z)
